raw_map/src/lib.rs

```diff
@@ -478,6 +448,39 @@ impl RawRoad {
             0
         }
     }
+
+    /// Returns the corrected (but untrimmed) center and total width for a road
+    pub fn untrimmed_road_geometry(&self) -> Result<(PolyLine, Distance)> {
+        let mut total_width = Distance::ZERO;
+        let mut sidewalk_right = None;
+        let mut sidewalk_left = None;
+        for l in &self.lane_specs_ltr {
+            total_width += l.width;
+            if l.lt.is_walkable() {
+                if l.dir == Direction::Back {
+                    sidewalk_left = Some(l.width);
+                } else {
+                    sidewalk_right = Some(l.width);
+                }
+            }
+        }
+
+        // If there's a sidewalk on only one side, adjust the true center of the road.
```

Comment:
I think placement tags would be applied somewhere around here

Reply:
I think that means that we should store a "true center" relative to the traffic divide line: `center_line_points`?. Placement tags describe the relationship between the osm geom and the lane markings and our job is to line markings up with each other, so lets pick a place that should always be unique on a road. The line dividing traffic is unique: the painted line on a standard road, down the middle of the median, on the inside edge of a oneway, where you would expect to see oncoming traffic, and down the middle of the "single twoway lane" situation for foot traffic and laneways (which is like where you would draw the dividing line if you were all on bikes).

With that reference, I can for e.g. find details about "the left sidewalk, after two lanes and a curb" without the other side of the road getting in the way. That way dual carriageways are treated the same as twoway roads, and we get to eliminate half of the possibility space by working with one direction at a time.

Reasoning about intersections (osm nodes, or rings in the graph) also doesn't care how the osm ways are split up, so normalising to a list of `HalfRoad`s inward and outward, would be a good first step. Then they are matched up in order to count center lines (if there are only 2, there isn't even any crossing traffic, so its a `Merge` not an `Intersection`, etc.), ...

Reply:
> The line dividing traffic is unique

Just to clarify a possible complication -- what about roads with two-way cycletracks on one side? 
![Screenshot from 2022-04-23 11-34-20](https://user-images.githubusercontent.com/1664407/164890964-54d31a89-f6e3-4b53-8006-e02386b28e01.png)
There are two lines where the direction of traffic changes (or even three -- the bollards between vehicles and cycletrack). I assume we'd use the line separating directions of vehicle traffic. If the cycletrack was a separate road to start, the center line there would be between the two directions of cycle traffic. And if we merged the two parallel ways, the vehicle center line would be take precedence.

> That way dual carriageways are treated the same as twoway roads

This is one reason I like this representation

Reply:
> > The line dividing traffic is unique
>
> There are two lines where the direction of traffic changes (or even three -- the bollards between vehicles and cycletrack).

You're right, what I described is very car-centric. That was my natural perspective given that I understand road tagging, but not cycletrack tagging. Having one "center line" is probably still a useful restriction to place one what a "road" can be, even if we need special support for this kind of situation.

My gut feeling is that treating the cycletrack as a "nested" road, or as a separate road that references the main road's geometry, might be an ergonomic way to work with them (thought not to implement), but I haven't thought too much about how cycletracks influence the transformations in unique ways.

Allowing backwards lanes on the forward side and vice versa, but still choose one center line to take precedence is probably the simplest way.